client/tests/integration/smartcontracts/executor_custom_data_model/src/multisig.rs

```diff
@@ -0,0 +1,24 @@
+//! Arguments to register and manage multisig account
+
+use alloc::{collections::btree_set::BTreeSet, vec::Vec};
+
+use iroha_data_model::{account::NewAccount, prelude::*};
+use serde::{Deserialize, Serialize};
+
+/// Arguments to multisig account register trigger
+#[derive(Serialize, Deserialize)]
+pub struct MultisigRegisterArgs {
+    // Account id of multisig account should be manually checked to not have corresponding private key (or having master key is ok)
+    pub account: NewAccount,
+    // List of accounts responsible for handling multisig account
+    pub signatories: BTreeSet<AccountId>,
+}
+
+/// Arguments to multisig account manager trigger
+#[derive(Serialize, Deserialize)]
+pub enum MultisigArgs {
+    /// Accept instructions proposal and initialize votes with the proposer's one
+    Instructions(Vec<InstructionBox>),
+    /// Accept vote for certain instructions
+    Vote(HashOf<Vec<InstructionBox>>),
```

Comment:
> You could have sent the same instruction repeatedly.

Yes, but than you have to store all this repetition on-chain which is not great.

We lenient on `creation_time` because we were merging transactions in the queue. 
Right not we don't have such problem, we might still want to de-duplicate payloads (i.e. multiple the same mints) in this case we might want to introduce nonce.
I would prefer to deal with it in separate PR if necessary.

Reply:
is there a reason to collect instructions and not the whole transaction?

Reply:
Currently you call trigger using ExecuteTrigger isi, you can't really pass transaction inside itself. 

Additionally i don't see a reason to pass transaction, since only part relevant to execution is payload.